links.go

```diff
@@ -198,8 +200,52 @@ func (nds ComponentDefinitions) validateLinks() error {
 			if !isLinkAllowed(componentName, targetName) {
 				return maskf(InvalidLinkDefinitionError, "invalid link to component '%s': component '%s' is not allowed to link to it", link.Component, componentName)
 			}
+
+			if err := nds.detectLinkCycle(link); err != nil {
+				return maskf(InvalidComponentDefinitionError, "invalid link to component '%s': %s", link.Component, err.Error())
+			}
+		}
+	}
+
+	return nil
+}
+
+// detectLinkCycle walks the links of the given component, looks up the target
+// components of each link, and follows this components links, until it finds a
+// loop. In case there is no loop, it does not throw an error. In case it
```

Comment:
Very minor: I would replace this with "In case it detects a loop, it returns an error, otherwise to returns nil".